### PR TITLE
Phase7 - 7.4 Add an API that can edit user’s posts.

### PR DESCRIPTION
**Pull request by kanghowoo**
Considering the cache data, I added the update board function.

### CODE REVIEW

File: src/main/java/com/mide/gangsaeng/board/BoardCacheRepositoryImpl.java [hunk 1]
```diff
@@ -39,6 +39,19 @@ public void write(Board board) {
...
+        try {
+            final long boardId = board.getId();
+            redisCommands.del(makeBoardKey(boardId));
+        } catch (RedisException e) {
+            log.error(e.getMessage(), e);
+        }
+
+        this.write(board);
+    }
+
```
Comment (TK-one) [inline src/main/java/com/mide/gangsaeng/board/BoardCacheRepositoryImpl.java:54]: I think we can just simply overwrite the cache value.

Comment (kanghowoo) [reply src/main/java/com/mide/gangsaeng/board/BoardCacheRepositoryImpl.java]: According to docs, if key already holds a value, it is overwritten, regardless of its type. 
So i changed.
[https://redis.io/docs/latest/commands/set/](url)

File: src/main/java/com/mide/gangsaeng/board/CacheableBoardRepository.java [hunk 2]
```diff
@@ -34,6 +33,15 @@ public void write(Board board) {
         cache.write(boardDataToBeStored);
     }
 
+    @Override
```
Comment (TK-one) [inline src/main/java/com/mide/gangsaeng/board/CacheableBoardRepository.java:36]: Question.
What will happen if we add transaction(`@Transactional`) for update here?
Give me the scenario

**Review by TK-one — APPROVED** [review]
LGTM

File: src/main/java/com/mide/gangsaeng/board/CacheableBoardRepository.java (hunk 2 above)
Comment (TK-one) [reply src/main/java/com/mide/gangsaeng/board/CacheableBoardRepository.java:36]: Then what will happen if we do not catch exceptions in write method for redis implementation?

> If we want to use transactions with a Redis client, we should use other commands or Spring Data Redis with additional settings.
What does it mean?

Comment (kanghowoo) [reply src/main/java/com/mide/gangsaeng/board/CacheableBoardRepository.java:36]: > Then what will happen if we do not catch exceptions in write method for redis implementation?
Stale data might be served to users.


> > If we want to use transactions with a Redis client, we should use other commands or Spring Data Redis with additional settings.
> 
> What does it mean?

It means if we want to use transactions in Redis(lettuce library), using only `@Transactional`annotation is not enough. For example, we can use `multi()` to enter transactions and `exec()` to execute all commands.
[https://lettuce.io/core/release/reference/index.html#_transactionsmulti](url)

In Spring Data Redis, we can use `@Transactional` after setting RedisTransactionManager or use `SessionCallback` with Redis commands.
[https://docs.spring.io/spring-data/redis/reference/redis/transactions.html](url)

**Pull request merged**